src/sentry/models/releases/set_commits.py

```diff
@@ -88,6 +94,45 @@ def set_commits_on_release(release, commit_list):
     latest_commit = None
     for idx, data in enumerate(commit_list):
         commit = set_commit(idx, data, release)
+
+        if idx == 0:
+            latest_commit = commit
+
+        commit_author_by_commit[commit.id] = commit.author
+        head_commit_by_repo.setdefault(data["repo_model"].id, commit.id)
+
+    release.update(
+        commit_count=len(commit_list),
+        authors=[
+            str(a_id)
+            for a_id in ReleaseCommit.objects.filter(
+                release=release, commit__author_id__isnull=False
+            )
+            .values_list("commit__author_id", flat=True)
+            .distinct()
+        ],
+        last_commit_id=latest_commit.id if latest_commit else None,
+    )
+    return head_commit_by_repo, commit_author_by_commit
+
+
+@metrics.wraps("set_commits_on_release")
+def set_commits_on_release_with_transaction_per_commit(release, commit_list):
+    # TODO(dcramer): would be good to optimize the logic to avoid these
+    # deletes but not overly important
+    ReleaseCommit.objects.filter(release=release).delete()
+
+    commit_author_by_commit = {}
+    head_commit_by_repo: dict[int, int] = {}
+
+    latest_commit = None
+    for idx, data in enumerate(commit_list):
+        with (
+            atomic_transaction(router.db_for_write(type(release))),
+            in_test_hide_transaction_boundary(),
+        ):
```

Comment:
The main difference with the original function is that this with block is added.

https://github.com/getsentry/sentry/blame/4639d43292bfe72866d7edbdcaa2bd28c7bdf914/src/sentry/models/releases/set_commits.py#L89-L92
<img width="319" alt="image" src="https://github.com/user-attachments/assets/5a974025-14e9-45c0-b678-95f593a83a44">